### DISCUSSION TITLE
Current Docker Images broken

### DISCUSSION THREAD

**Discussion by stackcoder** [firefly-iii/firefly-iii#8760] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [ ] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Container startup fails with `PHP Fatal error:  Uncaught ReflectionException: Class "view" does not exist` and stays in unhealthy state. As a workaround I downgraded the image back to `fireflyiii/core:version-6.0.30`, which starts correctly.

Also I was unable to find a Dockerfile or likely a specification, which makes debugging harder.

### Debug information

In Container.php line 904:

  Class "auth" does not exist

  [!] You can safely ignore the error about the 'fully qualified domain name'.

AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.23.0.4. Set the 'ServerName' directive globally to suppress this message
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.23.0.4. Set the 'ServerName' directive globally to suppress this message
[Thu Apr 04 10:39:14.691105 2024] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.56 (Debian) configured -- resuming normal operations
[Thu Apr 04 10:39:14.691222 2024] [core:notice] [pid 1] AH00094: Command line: 'apache2 -D FOREGROUND'
[Thu Apr 04 10:39:54.031196 2024] [php:error] [pid 108] [client 127.0.0.1:51496] PHP Fatal error:  Uncaught ReflectionException: Class "view" does not exist in /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php:904\nStack trace:\n#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(904): ReflectionClass->__construct()\n#1 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(787): Illuminate\\Container\\Container->build()\n#2 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(1026): Illuminate\\Container\\Container->resolve()\n#3 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(723): Illuminate\\Foundation\\Application->resolve()\n#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(1011): Illuminate\\Container\\Container->make()\n#5 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(1446): Illuminate\\Foundation\\Application->make()\n#6 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(237): Illuminate\\Container\\Container->offsetGet()\n#7 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(208): Illuminate\\Support\\Facades\\Facade::resolveFacadeInstance()\n#8 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(351): Illuminate\\Support\\Facades\\Facade::getFacadeRoot()\n#9 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/RegisterErrorViewPaths.php(16): Illuminate\\Support\\Facades\\Facade::__callStatic()\n#10 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(900): Illuminate\\Foundation\\Exceptions\\RegisterErrorViewPaths->__invoke()\n#11 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(875): Illuminate\\Foundation\\Exceptions\\Handler->registerErrorViewPaths()\n#12 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(806): Illuminate\\Foundation\\Exceptions\\Handler->renderHttpException()\n#13 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(697): Illuminate\\Foundation\\Exceptions\\Handler->prepareResponse()\n#14 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(586): Illuminate\\Foundation\\Exceptions\\Handler->renderExceptionResponse()\n#15 /var/www/html/app/Exceptions/Handler.php(178): Illuminate\\Foundation\\Exceptions\\Handler->render()\n#16 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(219): FireflyIII\\Exceptions\\Handler->render()\n#17 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(196): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->renderHttpResponse()\n#18 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(256): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleException()\n#19 [internal function]: Illuminate\\Foundation\\Bootstrap\\HandleExceptions->Illuminate\\Foundation\\Bootstrap\\{closure}()\n#20 {main}\n\nNext Illuminate\\Contracts\\Container\\BindingResolutionException: Target class [view] does not exist. in /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php:906\nStack trace:\n#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(787): Illuminate\\Container\\Container->build()\n#1 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(1026): Illuminate\\Container\\Container->resolve()\n#2 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(723): Illuminate\\Foundation\\Application->resolve()\n#3 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(1011): Illuminate\\Container\\Container->make()\n#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(1446): Illuminate\\Foundation\\Application->make()\n#5 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(237): Illuminate\\Container\\Container->offsetGet()\n#6 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(208): Illuminate\\Support\\Facades\\Facade::resolveFacadeInstance()\n#7 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(351): Illuminate\\Support\\Facades\\Facade::getFacadeRoot()\n#8 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/RegisterErrorViewPaths.php(16): Illuminate\\Support\\Facades\\Facade::__callStatic()\n#9 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(900): Illuminate\\Foundation\\Exceptions\\RegisterErrorViewPaths->__invoke()\n#10 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(875): Illuminate\\Foundation\\Exceptions\\Handler->registerErrorViewPaths()\n#11 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(806): Illuminate\\Foundation\\Exceptions\\Handler->renderHttpException()\n#12 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(697): Illuminate\\Foundation\\Exceptions\\Handler->prepareResponse()\n#13 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(586): Illuminate\\Foundation\\Exceptions\\Handler->renderExceptionResponse()\n#14 /var/www/html/app/Exceptions/Handler.php(178): Illuminate\\Foundation\\Exceptions\\Handler->render()\n#15 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(219): FireflyIII\\Exceptions\\Handler->render()\n#16 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(196): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->renderHttpResponse()\n#17 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(256): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleException()\n#18 [internal function]: Illuminate\\Foundation\\Bootstrap\\HandleExceptions->Illuminate\\Foundation\\Bootstrap\\{closure}()\n#19 {main}\n  thrown in /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php on line 906

### Expected behaviour

Container starts up correctly like previous versions did.
Developers can find `Dockerfile` and `entrypoint.sh`.

### Steps to reproduce

1. Update docker compose using pull and recreate

**Comment by JC5** [comment]
You can follow the links from [this repository](https://github.com/firefly-iii/docker) to find the Dockerfile(s) you need.

Having said that, `firefly-iii/core:latest` works fine for me. Is there a specific tag you believe is broken?

**Comment by stackcoder** [comment]
Thank you for providing Dockerfiles! Combining GitHub with Azure DevOps was unexpected for me, especially since there is ghcr.

I've tested with `:latest` and `:version-6.1.12` both were unable to start, than I rolled back to previous minor release.

**Comment by JC5** [comment]
Yes, which is why I don't put all my eggs in one basket. But alas, GitHub is owned by Microsoft. 

I'm getting very little feedback from others about these images, so I'm not really inclined to do a deep dive.

Have you considered deleting all your images and trying again?

**Comment by Nicxx2** [comment]
@stackcoder 
You can use this:
https://github.com/Nicxx2/fireflyiii-quickstart

This was created today using the firefly-iii:latest. Everything is already setup and you just have to pull the docker image and use the docker compose that is included in the description.

**Comment by sharmay** [comment]
My Environment:
FedoraCoreOS 41.20241122.3.0
podman version 5.3.0
podman running rootless
database postgres
Postgresq version : 17

I followed https://docs.firefly-iii.org/how-to/firefly-iii/installation/docker/#plain-docker

`php -v`
```
PHP 8.3.13 (cli) (built: Nov  1 2024 04:48:44) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.3.13, Copyright (c) Zend Technologies
    with Zend OPcache v8.3.13, Copyright (c), by Zend Technologies
```

## log
```
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
Generated optimized autoload files containing 7737 classes

In Container.php line 946:
  Target class [auth] does not exist.  
In Container.php line 944:
  Class "auth" does not exist  

  [i] Wait for the database. You may see an error about an 'aborted connection', this is normal.
wait-for-it.sh: waiting 60 seconds for postgresql:5432
wait-for-it.sh: postgresql:5432 is available after 0 seconds
  [✓] DB is up.
  [i] Wait another 10 seconds in case the DB needs to boot.
)  [✓] Done waiting for the DB to boot.

  [i] Will run database commands.

In Container.php line 946:
  Target class [auth] does not exist.  
In Container.php line 944:
  Class "auth" does not exist  
                               
  [i] Will run upgrade commands.

In Container.php line 946:
  Target class [auth] does not exist.  
In Container.php line 944:
  Class "auth" does not exist  

  [i] Will run report commands.

In Container.php line 946:
  Target class [auth] does not exist.  
In Container.php line 944:
  Class "auth" does not exist  
In Container.php line 946:
  Target class [auth] does not exist.  
In Container.php line 944:
  Class "auth" does not exist  
In Container.php line 946:
  Target class [auth] does not exist.  
In Container.php line 944:
  Class "auth" does not exist  
In Container.php line 946:
  Target class [auth] does not exist.  
In Container.php line 944:
  Class "auth" does not exist  
In Container.php line 946:
  Target class [auth] does not exist.  
In Container.php line 944:                            
  Class "auth" does not exist  
```

## I manually ran:

`php artisan firefly-iii:upgrade-database`

```
In Container.php line 946:
  Target class [auth] does not exist.  
In Container.php line 944:
  Class "auth" does not exist  
```

A connection error?

I used psql to verify that DB is working correctly in DB container with given DB_ settings.

When I try to access URL, I get similar errors like https://github.com/orgs/firefly-iii/discussions/8760#discussion-6470484

**Reply by JC5** [reply]
The current docker images use root (inside the container). Not something I'm particularly happy with, but that's the case right now. Other than that, I have no idea.
